### PR TITLE
Basic tests

### PR DESCRIPTION
**Pull request by plopesc**
Including basic tests described on my comment in #5.
Once #8 will be in, we should improve a bit PlugBasicTest

### CODE REVIEW

File: plug.test [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * @file
+ * Basic tests cases for plug module.
+ */
+use \Drupal\plug\Util\Module;
+
+class PlugBasicTest extends \DrupalWebTestCase {
```
Comment (e0ipso) [inline plug.test]: ~~Can we turn this into a unit test?~~

Comment (e0ipso) [inline plug.test]: ~~Maybe rename the class to `PlugBasicUnit`.~~

File: plug.info [hunk 2]
```diff
@@ -4,3 +4,5 @@ core = 7.x
 php = 5.4.2
 dependencies[] = composer_manager
 dependencies[] = xautoload (>= 7.x-5.0)
+
+files[] = plug.test
```
Comment (e0ipso) [inline plug.info]: ~~Rename to `PlugBasicUnit.test`?~~

Rename to `PlugBasicTest.test`?

File: modules/plug_example/plug_example.info [hunk 3]
```diff
@@ -3,3 +3,4 @@ description = A module example on how to use plugins.
 core = 7.x
 dependencies[] = plug
 
+files[] = plug_example.test
```
Comment (e0ipso) [inline modules/plug_example/plug_example.info]: Rename to `PlugExampleBasicTest.test`?

File: plug.test (hunk 1 above)
Comment (e0ipso) [inline plug.test]: Cannot have unit testing since we need to bootstrap for the list of modules. :weary:

File: plug.test [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      'shortcut' => 'modules/shortcut',
+      'system' => 'modules/system',
+      'taxonomy' => 'modules/taxonomy',
+      'text' => 'modules/field/modules/text',
+      'toolbar' => 'modules/toolbar',
+      'user' => 'modules/user',
+      'standard' => 'profiles/standard',
+    );
+
+    $this->assertEqual($expected_values, Module::getModuleDirectories());
```
Comment (e0ipso) [inline plug.test]: Will this fail in other Drupal installations with different enabled modules? Maybe we should just make sure that it contains:
- The required core modules.
- The module itself.
- The module dependencies.

File: plug.test [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      'Drupal\\shortcut' => 'modules/shortcut/src',
+      'Drupal\\system' => 'modules/system/src',
+      'Drupal\\taxonomy' => 'modules/taxonomy/src',
+      'Drupal\\text' => 'modules/field/modules/text/src',
+      'Drupal\\toolbar' => 'modules/toolbar/src',
+      'Drupal\\user' => 'modules/user/src',
+      'Drupal\\standard' => 'profiles/standard/src',
+    ));
+
+    $this->assertEqual($expected_values, Module::getNamespaces());
```
Comment (e0ipso) [inline plug.test]: Same as https://github.com/mateu-aguilo-bosch/plug/pull/10/files#r22309470

**Comment by e0ipso** [comment]
Now that we have tests, I'll write a PR to integrate with Travis.

File: plug.test (hunk 4 above)
Comment (plopesc) [inline plug.test]: Theoretically when you execute tests, you're always using the "testing" profile, so the list of enabled modules should be always the same.

What do you think?

Comment (e0ipso) [inline plug.test]: Yeah, the only problem that I see is that this is then coupled to the testing `setUp`.

**Comment by amitaibu** [comment]
nice stuff! :)

**Pull request merged**